src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas2/OffenderService.kt

```diff
@@ -66,17 +67,30 @@ class OffenderService(
       return ProbationOffenderSearchResult.NotFound(nomsNumber)
     } else {
       val probationOffenderDetail = probationOffenderDetailList[0]
+
+      // check for restrictions or exclusions
       val isLao = probationOffenderDetail.currentExclusion == true || probationOffenderDetail.currentRestriction == true
       if (isLao) {
         return ProbationOffenderSearchResult.Forbidden(nomsNumber)
       }
 
+      // check inmate details from Prison API
```

Comment:
As it stands we check for restrictions or exclusions on the offender then check to see if they are in the same prison.  Might be worth switching this order so that we aren't in a position to return a forbidden error for an offender with a restriction, that potentially is from a prison different to where the POM is based?

Reply:
hmm having looked at it again I'm wondering if LAO checks should go first actually - we already have the info from the Delius context on whether they are an LAO, so if they are we don't need to go and do the Prison API call and we can return early. AFAIK the controls on LAOs should apply even if the offender is within the same prison.